### ISSUE TITLE
lxml fails to install on Python 3.8 on native Windows

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#1980]
This happened on native Windows and on Python 3.8 in the Appveyor CI.

With latest package levels, lxml 4.4.2 installs fine:
https://ci.appveyor.com/project/andy-maier/pywbem/builds/29198716/job/fw1dsybwfc952d6x

With minimum package levels, lxml 4.2.4 fails to install:
https://ci.appveyor.com/project/andy-maier/pywbem/builds/29198716/job/uodow078jgli0k5u

Some errors that appear:
* ERROR: b"'xslt-config' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n"
  ** make sure the development packages of libxml2 and libxslt are installed **
* c:\users\appveyor\appdata\local\temp\1\pip-build-crx5y463\lxml\src\lxml\includes/etree_defs.h(14): fatal error C1083: Cannot open include file: 'libxml/xmlversion.h': No such file or directory
*  C:\Users\appveyor\AppData\Local\Temp\1\xmlXPathInitgvqjslda.c(1): fatal error C1083: Cannot open include file: 'libxml/xpath.h': No such file or directory
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\cl.exe' failed with exit status 2
  Could not find function xmlCheckVersion in library libxml2. Is libxml2 installed?

Note that libxml2 gets installed as part of the choco package "xsltproc", and that it works with latest package levels.

**Comment by andy-maier** [comment]
Here is what seems to be the significant difference between minimum and latest package installations, even when the minimum-constraints file specifies the latest version:

* Latest package level (no constraints file, requirements file specifies lxml>=4.4.2):
  From the [Appveyor run](https://ci.appveyor.com/project/andy-maier/pywbem/builds/29199502/job/h435803ud7jtvjw5):
  ```
  Collecting lxml>=4.4.2
    Downloading https://files.pythonhosted.org/packages/c2/06/b9694a71a29a35f4db71dd88cc91c54ca051f3ffabe96ff8454e1f2ee8ff/lxml-4.4.2-cp38-cp38-win_amd64.whl (3.7MB)
  ```

* Minimum package level (constraints file specifies lxml==4.4.2, requirements file specifies lxml>=4.4.2):
  From the [Appveyor run](https://ci.appveyor.com/project/andy-maier/pywbem/builds/29199502/job/bq6mwi086n9amuuq):
  ```
  Collecting lxml==4.4.2 (from -c minimum-constraints.txt (line 56))
    Downloading https://files.pythonhosted.org/packages/e4/19/8dfeef50623892577dc05245093e090bb2bab4c8aed5cad5b03208959563/lxml-4.4.2.tar.gz (2.6MB)
  ```

So when using the source tarball, the installation fails with missing include files and the like, while when using the wheel file, the binaries are already in the wheel file and the missing include files do not matter because no binary build happens.

**Comment by andy-maier** [comment]
It turns out that there is a fix for this in lxml in https://github.com/lxml/lxml/pull/291. It has been merged to master and should therefore be in the next version after the current version 4.4.2.

**Comment by andy-maier** [comment]
Coming up with a workaround does not seem to be easy. The trick that worked for issue #1975 which is to not specify lxml in the minimum constraints file to get pip to use the wheel file, does not work. Pip still downloads the source tarball in that case and then fails due to missing libxml2 headers:

* Works fine with latest package level (no constraints file, requirements file specifies lxml>=4.4.1):
  From the [Appveyor run](https://ci.appveyor.com/project/andy-maier/pywbem/builds/29201696/job/99pg8vb42b94133y):
  ```
  Collecting lxml>=4.4.1
  Downloading https://files.pythonhosted.org/packages/c2/06/b9694a71a29a35f4db71dd88cc91c54ca051f3ffabe96ff8454e1f2ee8ff/lxml-4.4.2-cp38-cp38-win_amd64.whl (3.7MB)
  ```

* Fails with minimum package level (constraints file does not specify lxml, requirements file specifies lxml>=4.4.1):
  From the [Appveyor run](https://ci.appveyor.com/project/andy-maier/pywbem/builds/29201696/job/q4x1smrcxm2hbj87):
  ```
  Collecting lxml>=4.4.1 (from -r dev-requirements.txt (line 29))
    Downloading https://files.pythonhosted.org/packages/e4/19/8dfeef50623892577dc05245093e090bb2bab4c8aed5cad5b03208959563/lxml-4.4.2.tar.gz (2.6MB)
  ```

**Comment by andy-maier** [comment]
The version of pip plays an important role. Pip 18.1.0 fails as described above, but pip 19.3.1 correctly downloads the lxml wheel file.

**Comment by andy-maier** [comment]
Addressed along with other issues, in PR #1976 for master, and TBD for 0.15.0.
